packages/fxa-auth-server/lib/routes/devices-and-sessions.js

```diff
@@ -254,12 +254,18 @@ module.exports = (log, db, config, customs, push, pushbox, devices, oauthdb) =>
       handler: async function (request) {
         log.begin('Account.deviceCommands', request);
 
-        const sessionToken = request.auth.credentials;
-        const uid = sessionToken.uid;
-        const deviceId = sessionToken.deviceId;
+        const credentials = request.auth.credentials;
+        const uid = credentials.uid;
+        const deviceId = credentials.deviceId;
         const query = request.query || {};
         const {index, limit} = query;
 
+        if (config.oauthDeviceCommandsEnabled === false) {
+          if (credentials.refreshTokenId) {
```

Comment:
How come the nested if here? I find it slightly more confusing than just using an &&. Also, shouldn't the refreshToken auth strategy take care of this?

Reply:
> shouldn't the refreshToken auth strategy take care of this?

No, because this one is specific to the use of device-commands rather than wholesale disablement of the entire devices API.